FogCreek/Glitch-Community
# quilled-tilapia: improve project page description metadata

## Links
* [quilled-tilapia.glitch.me](https://quilled-tilapia.glitch.me)
* [Manuscript: Better project/team/user/collection page structured data](https://glitch.fogbugz.com/f/cases/3327826/Better-project-team-user-collection-page-structured-data)

## GIF/Screenshots:

**Projects that use the provided description**
![Screen Shot 2019-05-29 at 11 37 22 AM](https://user-images.githubusercontent.com/7584833/58570537-242f3900-8206-11e9-9151-820d0bf486cf.png)

**Projects that use default text instead of the provided description**

Top to bottom:
1. no description
2. "A tidy project that does clean things"-style description
3. default description, remixed from one of our hello-* templates

![sidebyside(7)](https://user-images.githubusercontent.com/7584833/58570805-96078280-8206-11e9-9315-7b3a3ca39474.png)

✅ ~🔜 These projects will eventually say something like, "_Check out ~tidy-project on Glitch, the friendly community..._" Need to confirm the exact copy with Ezra when he's back from vacation.~ 

## Changes:
- Detect if a project is using a default description, and use default text for metadata description if so
- Add `constants.tagline`, use it instead of 'Glitch is the friendly...' string literals

## How To Test:
Use [metatags.io](https://metatags.io) to test the following URLs:

You should see "Check out ~domain on Glitch, the friendly...":
- https://quilled-tilapia.glitch.me/~glory-guilt
- https://quilled-tilapia.glitch.me/~wonderful-kick
- https://quilled-tilapia.glitch.me/~hello-webpage

You should see "[Project description] 🎏 Glitch is the friendly...":
- https://quilled-tilapia.glitch.me/~culture-zine

## Feedback I'm looking for:
Is it broken in any way that I missed

## Things left to do before deploying:
- [x] Talk to Ezra about exactly what text we want to use
- [x] Confirm that we're cool using default project avatars when a project doesn't define an avatar
